server/protocols/jmap-draft/src/main/java/org/apache/james/jmap/draft/methods/SetMailboxesUpdateProcessor.java

```diff
@@ -266,14 +269,20 @@ private void updateMailbox(Mailbox mailbox, MailboxUpdateRequest updateRequest,
                     .toMailboxAcl(),
                 mailboxSession);
         }
-        if (!originMailboxPath.equals(destinationMailboxPath)) {
-            mailboxManager.renameMailbox(mailbox.getId(), destinationMailboxPath, mailboxSession);
 
-            subscriptionManager.unsubscribe(mailboxSession, originMailboxPath.getName());
-            subscriptionManager.subscribe(mailboxSession, destinationMailboxPath.getName());
+        if (!originMailboxPath.equals(destinationMailboxPath)) {
+            List<MailboxRenamedResult> mailboxRenamedResults = mailboxManager.renameMailbox(mailbox.getId(), destinationMailboxPath, mailboxSession);
+            updateSubcription(mailboxRenamedResults, mailboxSession);
         }
     }
 
+    private void updateSubcription(List<MailboxRenamedResult> mailboxRenamedResults, MailboxSession mailboxSession) {
+        mailboxRenamedResults.forEach(Throwing.consumer(mailboxRenamedResult -> {
+            subscriptionManager.unsubscribe(mailboxSession, mailboxRenamedResult.getOriginPath().getName());
+            subscriptionManager.subscribe(mailboxSession, mailboxRenamedResult.getDestinationPath().getName());
```

Comment:
Given a non subscribed mailbox, do we create a sibscription for it upon rename?

If the mailbox is unsubscribed, renaming it should keep it unsubscribed.

Reply:
yes, it a corner case.I think we should refined this api.

Reply:
I guess when rename mailbox `A` to `B` is something like:
validate `B` >> create mailbox `B` >> update mailboxpath related to  `A` >> (delete mailbox `A` and handle unsub `A` and sub `B`) right?

If we do like:
 validate `B` >> create mailbox `B` + subscription `B` >> update mailboxpath related to  `A` >> unsub `A` and delete mailbox `A`.
we can get out of case unsub a ghost mailbox.It's ok?

Reply:
I don't understand your last message.

Reply:
we can talk tomorrow morning.